controllers/numaresourcesoperator_controller.go

```diff
@@ -155,6 +155,7 @@ func (r *NUMAResourcesOperatorReconciler) updateStatus(ctx context.Context, inst
 
 	if err := status.Update(ctx, r.Client, instance, condition, reason, message); err != nil {
 		klog.InfoS("Failed to update numaresourcesoperator status", "Desired condition", status.ConditionDegraded, "error", err)
+		r.Recorder.Eventf(instance, corev1.EventTypeWarning, "UpdateStatusFailed", "failed to update numaresourcesoperator status with desired condition: %s", condition)
```

Comment:
this doesn't deserve a event I believe, users can barely do anything about that.